### PR TITLE
Avoid to delete the file not on stage

### LINKED ISSUE #137: Missing updated file on stage when command fails
Missing updated file on stage when `qs` commands was failed.

Impact
- On not staged and modified files

Not Impact
- Untracked files

```bash
~/repo/src/github.com/kamontia/qs master* ⇣⇡ kamo@kamo
kamo@kamo: echo "Update something" >> version.go

~/repo/src/github.com/kamontia/qs master* ⇣⇡ kamo@kamo
kamo@kamo: git st
ブランチ master
Your branch and 'origin/master' have diverged,
and have 5 and 8 different commits each, respectively.
  (use "git pull" to merge the remote branch into yours)

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

        modified:   version.go

追跡されていないファイル:
  (use "git add <file>..." to include in what will be committed)

        somefile

no changes added to commit (use "git add" and/or "git commit -a")

~/repo/src/github.com/kamontia/qs master* ⇣⇡ kamo@kamo
kamo@kamo: ./qs -n 0..1 -f
{{} [0 1 0]} 0xc00007c420 0xc00009ea80
ERRO[0000] Error. QS try to recovery...                 
ERRO[0000] exit status 1                                
ERRO[0000] Completed. Please rebase manually.           

~/repo/src/github.com/kamontia/qs master* ⇣⇡ kamo@kamo
kamo@kamo: git st
ブランチ master
Your branch and 'origin/master' have diverged,
and have 5 and 8 different commits each, respectively.
  (use "git pull" to merge the remote branch into yours)

追跡されていないファイル:
  (use "git add <file>..." to include in what will be committed)

        somefile

nothing added to commit but untracked files present (use "git add" to track)
```

### PR DESCRIPTION
**Pull request by kamontia**
Avoid to delete the unstaged file.
@chaspy Please review and merge.

Close #137

### CODE REVIEW

**Comment by kamontia** [comment]
@chaspy I updated the test case for checking this issue works correctly. Please review and merge.

File: script/test.sh [hunk 1]
```diff
@@ -82,16 +82,19 @@ test_rebase_abort () {
    git add file-11
    git commit -m "Add file-11"
 
+   echo "Add some modification" >> file-11
+   HAS_UNMODIFIED_FILE=$(git status --short | grep file-11 | wc -l)
```
Comment (chaspy) [inline script/test.sh]: In the current implementation, `HAS_UNMODIFIED_FILE` has always "1" and the following condition `-a "${HAS_UNMODIFIED_FILE}" != "0"` returns always "true".

`HAS_UNMODIFIED_FILE=$(git status --short | grep file-11 | wc -l)` should be after `./"${EXEC_COMMAND}" -n 2..5 -f -d` line.

Comment (kamontia) [reply script/test.sh]: @chaspy You are right. I fixed it.

**Comment by kamontia** [comment]
@chaspy I fixed test case. Please review and marge.

**Comment by kamontia** [comment]
@chaspy Please review and merge.

We can confirm that `test_rebase_abort` is failed at 132a8ae.
CircleCI: https://circleci.com/gh/kamontia/qs/345?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

After fix bug, it is green at 632a44b

**Review by chaspy — APPROVED** [review]
LGTM 👍 qs will be more convenient!

**Pull request merged**